src/lib.rs

```diff
@@ -328,16 +332,48 @@ mod test {
         let bitcoind = BitcoinD::with_conf(&exe, &conf).unwrap();
         assert_eq!(bitcoind.client.get_peer_info().unwrap().len(), 0);
         let other_conf = Conf {
-            p2p: bitcoind.p2p_connect().unwrap(),
+            p2p: bitcoind.p2p_connect(false).unwrap(),
             ..Default::default()
         };
         let other_bitcoind = BitcoinD::with_conf(&exe, &other_conf).unwrap();
         assert_eq!(bitcoind.client.get_peer_info().unwrap().len(), 1);
         assert_eq!(other_bitcoind.client.get_peer_info().unwrap().len(), 1);
     }
 
-    fn init() -> String {
-        let _ = env_logger::try_init();
+    #[test]
+    fn test_multi_p2p() {
+        let conf_node1 = Conf {
+            p2p: P2P::Yes,
+            ..Default::default()
+        };
+        let node1 = BitcoinD::with_conf(exe_path(), &conf_node1).unwrap();
+
+        // Create Node 2 connected Node 1
+        let conf_node2 = Conf {
+            p2p: node1.p2p_connect(true).unwrap(),
+            ..Default::default()
+        };
+        let node2 = BitcoinD::with_conf(exe_path(), &conf_node2).unwrap();
+
+        // Create Node 3 Connected To Node 2
+        let conf_node3 = Conf {
+            p2p: node2.p2p_connect(false).unwrap(),
+            ..Default::default()
+        };
+        let node3 = BitcoinD::with_conf(exe_path(), &conf_node3).unwrap();
+
+        // Get each nodes Peers
+        let node1_peers = node1.client.get_peer_info().unwrap();
+        let node2_peers = node2.client.get_peer_info().unwrap();
+        let node3_peers = node3.client.get_peer_info().unwrap();
+
+        // Peers found
+        assert!(node1_peers.len() >= 1);
+        assert!(node2_peers.len() >= 1);
```

Comment:
nit:
```suggestion
        assert_eq!(node1_peers.len(), 2);
        assert_eq!(node2_peers.len(), 2);
```

Reply:
it's `>= 1` because the node may or may not connect to each other in time